### PR TITLE
[CI Environment] Implemented name shortening for TemplateSpecs artifacts & extended docs

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Implemented name shortening for TemplateSpecs artifacts
- Extended docs
- Removed some old code from the ADO Artifacts generation that would, for example, cut their name at 255 characters

Successfull RSV publishing with shortened name:
image

| Pipeline |
| - |
| [KeyVault: Vaults](https://github.com/Azure/ResourceModules/actions/workflows/ms.keyvault.vaults.yml) |
| [RecoveryServices: Vaults](https://github.com/Azure/ResourceModules/actions/workflows/ms.recoveryservices.vaults.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: docs/wiki/The CI environment - Publishing.md [hunk 1]
```diff
@@ -17,11 +18,76 @@ The publishing phase concludes each module's pipeline. If all previous tests suc
...
+## Module identifiers
+
+The names of published modules differ slighly depending on the location they are published to. This is rooted in the different requirements per target location. In the following you can find the rules applied for each:
+
+<details>
+<summary>Template Specs</summary>
+
+**Actions**
+
+- Make lowercase
```
Comment (eriqua) [inline docs/wiki/The CI environment - Publishing.md:34]:
```suggestion
- Get the relative path of the module template file parent folder
- Remove the root folder name `modules`
- Make lowercase
```

File: docs/wiki/The CI environment - Publishing.md [hunk 2]
```diff
@@ -17,11 +18,76 @@ The publishing phase concludes each module's pipeline. If all previous tests suc
...
+    - After: `ms.recoveryservices.vaults.replicationfabrics.replicationprotectioncontainers.mappings`
+
+</details>
+
+<details>
+<summary>Private Bicep Registry (Azure Container Registry)</summary>
+
+**Actions**
+
+- Make lowercase
```
Comment (eriqua) [inline docs/wiki/The CI environment - Publishing.md:55]:
```suggestion
- Get the relative path of the module template file parent folder
- Remove the root folder name `modules`
- Make lowercase
```

File: docs/wiki/The CI environment - Publishing.md [hunk 3]
```diff
@@ -17,11 +18,76 @@ The publishing phase concludes each module's pipeline. If all previous tests suc
...
+- Replace all `\` or `/` with `.`
+- Add the `bicep\modules` prefix
```
Comment (eriqua) [inline docs/wiki/The CI environment - Publishing.md:57]:
```suggestion
- Add the `bicep/modules` prefix
```

File: docs/wiki/The CI environment - Publishing.md [hunk 4]
```diff
@@ -17,11 +18,76 @@ The publishing phase concludes each module's pipeline. If all previous tests suc
...
+   - After: `bicep/modules/microsoft.recoveryservices.vaults.replicationfabrics.replicationprotectioncontainers.replicationprotectioncontainermappings`
+
+</details>
+
+<details>
+<summary>Azure DevOps Universal Packages</summary>
+
+**Actions**
+
+- Make lowercase
```
Comment (eriqua) [inline docs/wiki/The CI environment - Publishing.md:75]:
```suggestion
- Get the relative path of the module template file parent folder
- Remove the root folder name `modules`
- Make lowercase
```

File: docs/wiki/The CI environment - Publishing.md (hunk 1 above)
Comment (AlexanderSehr) [reply docs/wiki/The CI environment - Publishing.md:34]: I agree with the 2nd point, but would argue the first isn't really relevant, is it? I did not intend to describe the algorithm, but simply the transformational actions on the name.

File: docs/wiki/The CI environment - Publishing.md (hunk 2 above)
Comment (AlexanderSehr) [reply docs/wiki/The CI environment - Publishing.md:55]: Same comment as before

File: docs/wiki/The CI environment - Publishing.md (hunk 4 above)
Comment (AlexanderSehr) [reply docs/wiki/The CI environment - Publishing.md:75]: Same comment as before

File: docs/wiki/The CI environment - Publishing.md (hunk 1 above)
Comment (eriqua) [reply docs/wiki/The CI environment - Publishing.md:34]: Sounds good, but in that case I'd suggest to remove `/deploy.bicep` from all the 'Before' examples in the file

Comment (AlexanderSehr) [reply docs/wiki/The CI environment - Publishing.md:34]: checking...

Comment (AlexanderSehr) [reply docs/wiki/The CI environment - Publishing.md:34]: done

Comment (AlexanderSehr) [reply docs/wiki/The CI environment - Publishing.md:35]: Added the 2nd line as `- Remove the root folder name `modules` from provided module reference` to all 3 variants

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**